generator/transit_generator.cpp

```diff
@@ -1,16 +1,141 @@
 #include "generator/transit_generator.hpp"
 
+#include "generator/osm_id.hpp"
+
+#include "routing_common/transit_serdes.hpp"
+#include "routing_common/transit_types.hpp"
+
+#include "coding/file_container.hpp"
+#include "coding/file_name_utils.hpp"
+#include "coding/file_writer.hpp"
+
+#include "platform/platform.hpp"
+
+#include "base/checked_cast.hpp"
 #include "base/logging.hpp"
 #include "base/macros.hpp"
 
+#include "3party/jansson/src/jansson.h"
+
+using namespace platform;
+using namespace routing;
+using namespace routing::transit;
+using namespace std;
+
+namespace
+{
+/// \returns file name without extension(s) by a file path. That means it gets a file name by a full path and
+/// and removes from the file name the first dot and everything after the dot, if the file name contains dot(s).
+/// For example,
+/// GetFileName("Russia_Nizhny Novgorod Oblast.transit.json") returns "Russia_Nizhny Novgorod Oblast"
+/// GetFileName("Russia_Nizhny Novgorod Oblast.mwm") returns "Russia_Nizhny Novgorod Oblast"
+string GetFileName(string const & filePath)
+{
+  string name = filePath;
+  my::GetNameFromFullPath(name);
+
+  string nameWithExt;
+  do
+  {
+    nameWithExt = name;
+    my::GetNameWithoutExt(name);
+  }
+  while (nameWithExt != name);
+
+  return name;
+}
+
+/// \brief Reads from |root| (json) and serializes an array to |serializer|.
+template <class Item>
+void SerializeObject(my::Json const & root, string const & key, Serializer<FileWriter> & serializer)
+{
+  vector<Item> items;
+
+  DeserializerFromJson deserializer(root.get());
+  deserializer(items, key.c_str());
+  serializer(items);
+}
+}  // namespace
+
 namespace routing
 {
 namespace transit
 {
-void BuildTransit(std::string const & mwmPath, std::string const & transitDir)
+// DeserializerFromJson ---------------------------------------------------------------------------
+void DeserializerFromJson::operator()(m2::PointD & p, char const * name)
 {
-  LOG(LINFO, ("mwm path:", mwmPath, ", directory with transit:", transitDir));
+  json_t * pointItem = my::GetJSONObligatoryField(m_node, name);
+  CHECK(json_is_object(pointItem), ());
+  FromJSONObject(pointItem, "x", p.x);
+  FromJSONObject(pointItem, "y", p.y);
+}
+
+void BuildTransit(string const & mwmPath, string const & transitDir)
+{
+  LOG(LERROR, ("This method is under construction and should not be used for building production mwm "
```

Comment:
LCRITICAL?

Reply:
Yes, but if I do that I'll have to remove the line while working on the section. So I'll be easier for me to leave as is.